pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -308,15 +296,47 @@ func getPinnedLibraries(registry string) []libInfo {
 	return res
 }
 
-// getLibrariesLanguageDetection runs process language auto-detection and returns languages to inject for APM Instrumentation.
-// The languages information is available in workloadmeta-store and attached on the pod's owner.
-func (w *Webhook) getLibrariesLanguageDetection(pod *corev1.Pod) []libInfo {
-	if config.Datadog().GetBool("admission_controller.auto_instrumentation.inject_auto_detected_libraries") {
-		// Use libraries returned by language detection for APM Instrumentation
-		return w.getAutoDetectedLibraries(pod)
-	}
+type libInfoLanguageDetection struct {
+	libs    []libInfo
+	enabled bool
 
-	return nil
+	injected bool
+}
+
+func (l *libInfoLanguageDetection) podMutator(v version) podMutator {
+	return podMutatorFunc(func(pod *corev1.Pod) error {
+		if l == nil || len(l.libs) == 0 || !v.usesInjector() || l.injected {
+			return nil
+		}
+
+		var libs []string
+		for _, lib := range l.libs {
+			libs = append(libs, string(lib.lang))
+		}
+
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGES_DETECTED",
+			Value: strings.Join(libs, ","),
+		})
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGE_DETECTION_ENABLED",
+			Value: strconv.FormatBool(l.enabled),
+		})
+
+		return nil
+	})
+}
+
+// getLibrariesLanguageDetection runs process language auto-detection and returns languages
```

Comment:
small nit:

This part doesn't run process language auto-detection.

Process language auto-detection runs in the process check/process collector. 

A more accurate description is:

`getLibrariesLanguageDetection returns the languages that were detected by process language detection`